homeassistant/components/lock/__init__.py

```diff
@@ -58,7 +62,13 @@ class LockEntityFeature(IntFlag):
 # Please use the LockEntityFeature enum instead.
 SUPPORT_OPEN = 1
 
-PROP_TO_ATTR = {"changed_by": ATTR_CHANGED_BY, "code_format": ATTR_CODE_FORMAT}
+PROP_TO_ATTR = {
+    "changed_by": ATTR_CHANGED_BY,
+    "code_format": ATTR_CODE_FORMAT,
+    "is_jammed": ATTR_IS_JAMMED,
+    "is_locked": ATTR_IS_LOCKED,
+    "is_locking": ATTR_IS_LOCKING,
```

Comment:
This is part of the state already, why add them as attributes?

Please note, this is a change to an entity component, which generally requires architectural discussions prior to making the change.

Reply:
When a lock is `jammed` we want to `know` in what state this happened,

Reply:
> Please note, this is a change to an entity component, which generally requires architectural discussions prior to making the change.

In the entity model all these attributes are available.

Reply:
> When a lock is jammed we want to know in what state this happened

That is a historical value? As it is jammed it is impossible to know if it is locked or not. However, that said, that would be nice to discuss architecturally.

> In the entity model all these attributes are available.

They are not exposed as attributes as they are not considered a state on their own.